### PR TITLE
Add ItemStack#displayName to get the formatted display name of an ItemStack

### PR DESCRIPTION
**Pull request by jpenilla**
The display name includes formatting such as item hover event, display name (in italics if renamed in anvil), and color based on rarity.

This is what vanilla uses for give command feedback, and when an item is shown in a death message.

### CODE REVIEW

File: Spigot-API-Patches/0005-Adventure.patch [hunk 1]
```diff
@@ -2869,10 +2869,19 @@ index 4ff149fd98895dd8ba45939a37c223b1f8d7281f..3dba4c361993e143e511b4f108ac0b44
...
 +    net.kyori.adventure.text.event.HoverEvent<net.kyori.adventure.text.event.HoverEvent.ShowItem> asHoverEvent(final @NotNull ItemStack item, final @NotNull java.util.function.UnaryOperator<net.kyori.adventure.text.event.HoverEvent.ShowItem> op);
++
++    /**
++     * Get the formatted display name of the {@link ItemStack}.
++     *
++     * @param itemStack the {@link ItemStack}
++     * @return display name of the {@link ItemStack}
++     */
++    @NotNull
++    net.kyori.adventure.text.Component displayName(@NotNull ItemStack itemStack);
```
Comment (Proximyst) [inline Spigot-API-Patches/0005-Adventure.patch:2880]: Always found it peculiar that we use JB annotations when Adventure uses Checker framework. Don't gotta change it, just something we should probably consider fixing later on

**Review by Proximyst — APPROVED** [review]
LGTM

File: Spigot-API-Patches/0005-Adventure.patch (hunk 1 above)
Comment (electronicboy) [reply Spigot-API-Patches/0005-Adventure.patch:2880]: Spigot used JBs when they started annotating stuff, we where still using javax and where discussing on where we wanted to go given that that was gonna break

Comment (jpenilla) [reply Spigot-API-Patches/0005-Adventure.patch:2880]: The `AnnotationTest` has been updated to allow checker annotations for return types, but not for parameter types. I looked into updating the test, but it seemed non-trivial. I just went with the JetBrains annotations here so we wouldn't have two types of annotations on the same method.

**Pull request merged**